pkg/perf/images/create_images.go

```diff
@@ -0,0 +1,199 @@
+package images
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	godigest "github.com/opencontainers/go-digest"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func CreateImages(parallelImagesNumber int) error {
+	pushedImages := GetImagesName()
+	for i := 0; i < len(pushedImages); i++ {
+		imageDir := fmt.Sprintf(imageDirPath, pushedImages[i])
+		if _, err := os.Stat(imageDir); os.IsNotExist(err) {
+			if err := MakeImageDir(imageDir); err != nil {
+				return err
+			}
+
+			if err := CreateLayout(imageDir); err != nil {
```

Comment:
No need to create the full layout. Just create blobs of various sizes. Right now, if i understand your code correctly, you are only creating one size. You just need a small range of blob sizes which you can mix and match when uploading.

Reply:
Oh wait! You are doing this so that you don't have to compute the shasums, sure, your layout possibilities are potentially very large. Would rather have a small set of blobs (so disk storage is minimal) and then precompute/store shasums and then create tests and combine on the fly.